Frameworks/QuartzCore/CALayer.mm

```diff
@@ -1570,9 +1568,9 @@ - (CATransform3D)sublayerTransform {
 */
 - (void)setBackgroundColor:(CGColorRef)color {
     if (color != nil) {
-        priv->backgroundColor = *[static_cast<UIColor*>(color) _getColors];
+        priv->backgroundColor = *(reinterpret_cast<const __CGColorQuad*>(CGColorGetComponents(color)));
```

Comment:
This is already assigning into a new struct, so it wouldn't be costly to take the extra safety of constructing a __CGColorQuad with the pointer and moving it